apps/openmw/mwrender/animation.cpp

```diff
@@ -113,19 +120,89 @@ namespace
             texEnv->setOperand2_RGB(osg::TexEnvCombine::SRC_COLOR);
 
             stateset->setTextureAttributeAndModes(mTexUnit, texEnv, osg::StateAttribute::ON);
+
+            // Reduce the texture list back down by one when a temporary glow finishes
+            // to allow FindLowestUnusedTexUnitVisitor to choose the same texunit again.
+            if (mDone)
+                stateset->getTextureAttributeList().resize(stateset->getTextureAttributeList().size() - 1);
         }
 
         virtual void apply(osg::StateSet *stateset, osg::NodeVisitor *nv)
         {
+            if (mDone)
+                return;
+
+            // If there is a permanent enchantment glow already on this object, give the
+            // permanent glow a pointer to a temporary glow added in a nested update callback
+            // The permanent glow will remove the temporary glow when it finishes, allowing
+            // the permanent glow to display its glow texture cycling properly.
+            if (mWatchedSpellGlow != NULL && mWatchedSpellGlow->isDone())
+            {
+                mNode->removeUpdateCallback(mWatchedSpellGlow);
+                mWatchedSpellGlow = NULL;
+            }
```

Comment:
In the new implementation, there shouldn't eve be more than one glowupdater applied at a time. In the case where there is already a glowupdater because of the object being an enchanted item, I temporarily change the color of the glow, as you suggest.